### PR TITLE
Refactor file interface to keep data and geometry consistently partitioned.

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: app/v1/dist/index.html [hunk 1]
```diff
@@ -119,13 +117,13 @@
               cube({ size: 1, center: true }).translate([0.5, 0.5, 0.5]).as('b'),
               cube({ size: 1, center: true }).translate([0.5, 0, 0]).as('c').material('metal'))
 
-writePaths({ path: 'assembly' },
-           x.toPaths({ tags: ['a'] }),
-           x.toPaths({ tags: ['b'] }),
-           x.toPaths({ tags: ['c'] }))
+writeStl({ path: 'assembly' },
+         x.toSolid({ tags: ['a'] }),
```
Comment (BarbourSmith) [inline app/v1/dist/index.html]: I like this notation better, it makes more sense to me at a glance

File: sys/writeFileSyncBrowser.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 const { getFile } = require('./files');
 
-const writeFileSync = (path, data, options) => {
+const writeFileSync = (path, data, options = {}) => {
   const file = getFile(path);
   file.data = data;
   for (const watcher of file.watchers) {
-    watcher(file);
+    watcher(file, options);
```
Comment (BarbourSmith) [inline sys/writeFileSyncBrowser.js]: 👍

**Comment by BarbourSmith** [comment]
Looks good to me 👍  👍

File: app/v1/dist/index.html (hunk 1 above)
Comment (pentacular) [reply app/v1/dist/index.html]: Cool.

Although I hope that the x.toSolid() stuff can be made mostly invisible to end-users at some point by adding some nice API support. :)

**Pull request merged**